File: Algorithms/Implementation/Grading Students/x.lua


--
-- Complete the 'gradingStudents' function below.
--
-- The function is expected to return an INTEGER_ARRAY.
-- The function accepts INTEGER_ARRAY grades as parameter.
--

local function processOne(grade)
    if grade < 38 then
        return grade
    else
        local remainder = grade % 5
        if remainder >= 3 then
            return grade + (5-remainder)
        else
            return grade
        end
    end
end

function gradingStudents(grades)
    local ret = {}
    for _, grade in ipairs(grades) do
        table.insert(ret, processOne(grade))
    end
    return ret
end

File: Algorithms/Implementation/Grading Students/x.erl

%
% Complete the 'gradingStudents' function below.
%
% The function is expected to return an INTEGER_ARRAY.
% The function accepts INTEGER_ARRAY grades as parameter.
%
processOne(Grade) when Grade < 38 -> Grade;
processOne(Grade) ->
    Remainder = Grade rem 5,
    if
        Remainder >= 3 ->
            Grade + (5-Remainder);
        true ->
            Grade
    end.

gradingStudents([]) -> [];
gradingStudents([Grade|Rest]) ->
    processOne(Grade) ++ gradingStudents(Rest).
